### PR TITLE
Add frontend changes for supporting a local DNS server

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [X] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#git-commit---signoff))

---

**What does this PR aim to accomplish?:**
Add support for a local DNS resolver in the *FTL*DNS beta testing branch

**How does this PR accomplish the above?:**
Implement necessary front-end changes to the Settings page
screenshot

**What documentation changes (if any) are needed to support this PR?:**
None

### CODE REVIEW

**Comment by DL6ER** [comment]
Updated screenshot
screenshot at 2018-04-18

File: settings.php [hunk 1]
```diff
@@ -205,6 +205,16 @@
...
+
+if (isset($setupVars["LOCAL_DNS_PORT"])) {
+    if($setupVars["LOCAL_DNS_PORT"]) {
+        $localDNS = $setupVars["LOCAL_DNS_PORT"];
+    } else {
+        $localDNS = false;
+    }
+} else {
+    $localDNS = false;
+}
```
Comment (Mcat12) [inline settings.php]: These `if` blocks can be flattened into one `if`

File: settings.php [hunk 2]
```diff
@@ -205,6 +205,16 @@
 } else {
     $piHoleLogging = true;
 }
+
+if (isset($setupVars["LOCAL_DNS_PORT"])) {
+    if($setupVars["LOCAL_DNS_PORT"]) {
+        $localDNS = $setupVars["LOCAL_DNS_PORT"];
```
Comment (Mcat12) [inline settings.php]: Sanitize the setting here, because we directly echo it out later.

File: scripts/pi-hole/php/savesettings.php [hunk 3]
```diff
@@ -198,9 +198,31 @@ function readAdlists()
...
+						// Remove entry without valid port
+						exec("sudo pihole -a localdnsport 0");
+					}
+				}
+				else
+				{
+					// Remove possible entry if unticked
+
+					exec("sudo pihole -a localdnsport 0");
+				}
```
Comment (Mcat12) [inline scripts/pi-hole/php/savesettings.php]: These `if` blocks can be flattened into one `if` block

**Pull request merged**